### PR TITLE
Fixes to HTTP implementation + unit tests.

### PR DESCRIPTION
**Pull request by kacf**
The last commit is all the unit tests, which is very long and repetitive. I don't blame you for not going through that in detail. 🙂

### CODE REVIEW

File: common/http/platform/beast/http.cpp [hunk 1]
```diff
@@ -176,6 +176,11 @@ void Client::ConnectHandler(const error_code &err, const asio::ip::tcp::endpoint
 
 	http_request_ = make_shared<http::request<http::buffer_body>>(
 		MethodToBeastVerb(request_->method_), request_->address_.path, BeastHttpVersion);
+
+	for (auto header : request_->headers_) {
```
Comment (oleorhagen) [inline common/http/platform/beast/http.cpp:201]: S: `const auto & header`

File: common/http/platform/beast/http.cpp [hunk 2]
```diff
@@ -722,7 +722,11 @@ error::Error Server::AsyncServeUrl(
 	const string &url, RequestHandler header_handler, RequestHandler body_handler) {
 	auto err = BreakDownUrl(url, address_);
 	if (err) {
-		return MakeError(InvalidUrlError, "Could to server URL " + url + ": " + err.String());
+		return MakeError(InvalidUrlError, "Could not parse URL " + url + ": " + err.String());
+	}
+
+	if (address_.path.size() > 0 && address_.path != "/") {
```
Comment (oleorhagen) [inline common/http/platform/beast/http.cpp:869]: Remind me, is `path.size() == 0` when it's the empty string? or can the `path` be unset?

**Review by oleorhagen — APPROVED** [review]
Holy

moly

File: common/http/platform/beast/http.cpp (hunk 2 above)
Comment (kacf) [reply common/http/platform/beast/http.cpp:869]: No, it can not be unset, they are like Golang strings. So what I wrote is identical to `path != ""`.

Comment (oleorhagen) [reply common/http/platform/beast/http.cpp:869]: Wouldn't this be equivalent to `address_.path != "/"` then?

Comment (kacf) [reply common/http/platform/beast/http.cpp:869]: No, because it needs to be not equal to both. Not equal to empty, and not equal to `/`.

Comment (oleorhagen) [reply common/http/platform/beast/http.cpp:869]: Then I would vote for `adress_path != "" && `

**Pull request merged**